fission-cli/library/Fission/CLI/Key/Store.hs

```diff
@@ -67,6 +68,26 @@ persist keyRole key = do
   path <- KeyStore.getPath keyRole
   forceWrite path $ B64.toByteString key
 
+fromFile ::
+  ( MonadIO m
+  , MonadKeyStore m key
+  , MonadRaise    m
+  , MonadLogger m
+  , m `Raises` Key.Error
+  )
+  => Proxy key
+  -> FilePath
+  -> m ()
+fromFile keyRole keyFile = do
+  scrubbed <- doesFileExist keyFile >>= \case
+    False ->
+      raise Key.DoesNotExist
+    True -> do
+      bs <- readFileBinary keyFile
+      return $ B64.Scrubbed.scrub bs
+  secretKey <- ensureM $ parse keyRole scrubbed
+  persist keyRole secretKey
```

Comment:
No need to change, but another option that flows in one direction (`<-`, `->` and `>>=` together is fine, but sometimes less load to just read one lines and arrows one way. Plus it subordinates detail)

```haskell
fromFile keyRole keyFile = do
  scrubbed  <- getScrubbed keyFile
  secretKey <- ensureM $ parse keyRole scrubbed
  persist keyRole secretKey
  where
    getScrubbed keyFile =
      doesFileExist keyFile >>= \case
        False ->
          raise Key.DoesNotExist

        True -> do
          bs <- readFileBinary keyFile
          return $ B64.Scrubbed.scrub bs
```

Reply:
done and learned a new thing (`forall`) in the process! 🙏